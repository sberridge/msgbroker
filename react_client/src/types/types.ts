export type publisherDetails = {
    id: string
    name: string
}

export type appPage = {
    title: string
    key: string
}

export type authedUser = {
    id: string
    name: string
}

export type statusMessage = {
    message: string
    type: string
}

export type formValidation = {[key:string]:string}

export type formValidationResponse =  [boolean, formValidation]