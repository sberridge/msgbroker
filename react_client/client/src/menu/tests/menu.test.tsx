import React from "react";
import renderer, { ReactTestRendererJSON } from "react-test-renderer";
import Menu from "../menu";
import {appPage} from "../../types/types"

type props = {
    onPageChange:(page:string)=>void
    currentPage:string
    pages:appPage[]
}
const createInstance = (props:props)=>{
    return <Menu
        onPageChange={props.onPageChange}
        currentPage={props.currentPage}
        pages={props.pages}
    ></Menu>
}

it("should render",()=>{
    let currentPage = "subscriptions";
    const onPageChange = (page:string)=>{
       comp.update(createInstance({
           onPageChange:onPageChange,
           currentPage:page,
           pages:appPages
       }))
    }
    const appPages:appPage[] = [
        {
            title:"Publishers",
            key: "publishers"
        },
        {
            title: "Subscriptions",
            key: "subscriptions"
        }
    ];
    const comp = renderer.create(
        createInstance({
            onPageChange:onPageChange,
            currentPage:currentPage,
            pages:appPages
        })
    )

    const getNestedChild = (el:ReactTestRendererJSON, levels:number) : ReactTestRendererJSON => {
        if(levels > 0) {
            if(el.children) {
                return getNestedChild(el.children[0] as ReactTestRendererJSON, --levels);
            }
        }
        return el;
    }

    let tree = comp.toJSON() as ReactTestRendererJSON;
    expect(tree).toMatchSnapshot();
    renderer.act(()=>{
        let a = getNestedChild(tree, 3);
        a.props.onClick();
    })

    tree = comp.toJSON() as ReactTestRendererJSON;
    expect(tree).toMatchSnapshot();

})
