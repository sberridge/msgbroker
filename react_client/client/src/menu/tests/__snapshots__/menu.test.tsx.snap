// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div
  className="tabs"
>
  <ul>
    <li
      className=""
    >
      <a
        onClick={[Function]}
      >
        Publishers
      </a>
    </li>
    <li
      className="is-active"
    >
      <a
        onClick={[Function]}
      >
        Subscriptions
      </a>
    </li>
  </ul>
</div>
`;

exports[`should render 2`] = `
<div
  className="tabs"
>
  <ul>
    <li
      className="is-active"
    >
      <a
        onClick={[Function]}
      >
        Publishers
      </a>
    </li>
    <li
      className=""
    >
      <a
        onClick={[Function]}
      >
        Subscriptions
      </a>
    </li>
  </ul>
</div>
`;
